crates/ide-db/src/imports/import_assets.rs

```diff
@@ -536,6 +537,29 @@ fn trait_applicable_items(
         required_assoc_items.insert(assoc);
         Some(assoc_item_trait.into())
     })
+    .filter(|candidate_trait_id| {
+        // we care about the following cases:
+        // 1. Trait's definition crate
+        // 2. Definition crates for all trait's generic arguments
+        //     a. This is recursive for fundamental types: `Into<Box<A>> for ()`` is OK, but `Into<Vec<A>> for ()`` is *not*.
+        // 3. Receiver type definition crate
+        //    a. This is recursive for fundamental types
+        let candidate_trait_id: TraitId = *candidate_trait_id;
```

Comment:
(style nit) Could be replaced by changing the closure param to `|&candidate_trait_id|`

Reply:
done!